runtime/src/main/java/org/corfudb/protocols/wireprotocol/orchestrator/HealNodeRequest.java

```diff
@@ -0,0 +1,74 @@
+package org.corfudb.protocols.wireprotocol.orchestrator;
+
+import static org.corfudb.protocols.wireprotocol.orchestrator.OrchestratorRequestType.HEAL_NODE;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+import lombok.Getter;
+
+/**
+ * An orchestrator request to heal an existing node to the cluster.
+ *
+ * @author Maithem
+ */
+public class HealNodeRequest implements CreateRequest {
+
+    @Getter
+    public String endpoint;
+
+    @Getter
+    boolean layoutServer;
+
+    @Getter
+    boolean sequencerServer;
+
+    @Getter
+    boolean logUnitServer;
+
+    @Getter
+    int stripeIndex;
+
+    public HealNodeRequest(String endpoint,
+                           boolean layoutServer,
+                           boolean sequencerServer,
+                           boolean logUnitServer,
+                           int stripeIndex) {
+        this.endpoint = endpoint;
+        this.layoutServer = layoutServer;
+        this.sequencerServer = sequencerServer;
+        this.logUnitServer = logUnitServer;
+        this.stripeIndex = stripeIndex;
+    }
+
+    public HealNodeRequest(byte[] buf) {
+        ByteBuffer byteBuffer = ByteBuffer.wrap(buf);
```

Comment:
Why are we manually dealing with ByteBuffers for string manipulation (on a non-performance critical path)? We have a JSON serializer we use for layouts, use it

Reply:
Fixed